webrender/src/prim_store.rs

```diff
@@ -2211,7 +2211,8 @@ impl Primitive {
 
             match segment_clip_chain {
                 Some(segment_clip_chain) => {
-                    if segment_clip_chain.clips_range.count == 0 {
+                    if segment_clip_chain.clips_range.count == 0 ||
+                       (!segment.may_need_clip_mask && !segment_clip_chain.has_non_local_clips) {
```

Comment:
hmm, shouldn't this be `segment_clip_chain.clips_range.count == 0 || !segment.may_need_clip_mask || segment_clip_chain.has_non_local_clips`?

Reply:
I don't think so.

The ```segment.may_need_clip_mask``` field (perhaps confusingly) refers to whether that segment might need a clip mask due to it's local segment.

So this condition says a segment can be drawn opaque if there are no relevant clip masks at all, or if the segment builder didn't have any local clips and there's no global clips that affect this segment.

It is subtle though, so I could be making a mistake here - could you explain your reasoning why it should be the above?

Reply:
I think `may_need_clip_mask` just needs to be renamed for clarity. The code is good.